libevmasm/GasMeter.cpp

```diff
@@ -271,6 +271,7 @@ unsigned GasMeter::runGas(Instruction _instruction, langutil::EVMVersion _evmVer
 	case Tier::Mid:     return GasCosts::tier4Gas;
 	case Tier::High:    return GasCosts::tier5Gas;
 	case Tier::Ext:     return GasCosts::tier6Gas;
+	case Tier::Special:  return GasCosts::tier6Gas; // FIXME A quick hack to get remix to work on tload
```

Comment:
BTW, these tiers seem to come from the [Yellow Paper](https://github.com/ethereum/yellowpaper). YP is no longer updated and now we should find these in the [execution specs](https://ethereum.github.io/execution-specs/). These tiers seem to have become more complicated over time and now many instructions have a tier of their own. I think we should follow the same convention here. I'd not change the existing instructions for now (maybe we'll refactor it some day), but for new instructions I'd introduce the tiers from the spec.

Specs for cancun have not been merged yet, but I found [the cancun gas tiers in the `forks/cancun` branch](https://github.com/ethereum/execution-specs/blob/forks/cancun/src/ethereum/cancun/vm/gas.py). I think that `TSTORE`/`TLOAD` would go into `GAS_WARM_ACCESS`, so we should introduce `Tier::WarmAccess` here.

Also, would be nice to add a comment explaining where the tiers come from. It was not obvious to me and I only realized this when reading the `MCOPY` EIP when it referred to the tiers.

Reply:
By the way, this change should also be covered with a gas test (`test/libsolidity/gasTests/`).

Reply:
Ok, for now I am using `Tier::WarmAccess`, which I had to introduce and `Gascosts::warmStorageReadCost`, which was already present in the code.

Reply:
Sounds good. Maybe it's actually better to do that in your PR and I'll do just `MCOPY`. Transient storage opcodes only have a static cost, which makes them simpler to handle than I thought. For `MCOPY` it's more complicated because most of the cost is dynamic and `GasMeter` has extra logic to handle such cases.

In any case, I can also take care of adding comments on how it works that I requested. Turns out that the tiers are only loosely based on Yellow Paper.

Reply:
#14772.